worlds/yugioh06/utils.py

```diff
@@ -0,0 +1,27 @@
+# support for AP world
+import io
+import pathlib
+import pkgutil
+import sys
+
+is_apworld = ".apworld" in sys.modules[__name__].__file__
+
+
+def get_zip_file():
+    filename = sys.modules[__name__].__file__
+    apworld_ext = ".apworld"
+    zip_path = pathlib.Path(filename[: filename.index(apworld_ext) + len(apworld_ext)])
+    return pkgutil.get_data(__name__, zip_path)
```

Comment:
Ah, I should've been more specific.

The point behind using pkgutil is that you can get rid of **all of this**, you don't need `isApWorld`, you don't need `get_zip_file` and your open_file function can now just be

```
def open_file(resource: str) -> bytes:
    parent_folder = pathlib.Path(__file__).stem
    return pkgutil.get_data(parent_folder, resource)
```

because pkgutil has an internal system that just deals with .zip files for you.

But also, do you actually even use this function anywhere? I couldn't find a single use of `open_file` in your entire apworld. If you're not using it anyway, you could get rid of this whole file, no?

Reply:
you're right I don't actually use this file anymore.